GANESH SONKAR ASSIGNMENT 3

Q1. A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

In [2]:
import pandas as pd

In [5]:
import numpy as np

In [6]:
import scipy

In [7]:
from scipy import stats

In [8]:
data=pd.read_csv("C:\\Users\\GANESH\\Downloads\\Cutlets.csv")

In [9]:
data

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [13]:
data.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [23]:
data.isnull().sum()

Unit A    0
Unit B    0
dtype: int64

In [26]:
data

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [27]:
unitB=data.iloc[:,1]

In [28]:
unitB

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [29]:
unitA=data.iloc[:,0]

In [30]:
unitA

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [31]:
p=stats.ttest_ind(unitA,unitB)

In [32]:
p

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

Inferences= (p value=0.472) > (α=0.05) Means we use Null Hypothesis i.e.

No difference between diameter cutlets of two units.


Q2.A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.

Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

In [33]:
Lab=pd.read_csv("C:\\Users\\GANESH\\Downloads\\LabTAT.csv")

In [59]:
Lab

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [60]:
P_value=stats.f_oneway(Lab.iloc[:,0],Lab.iloc[:,1],Lab.iloc[:,2],Lab.iloc[:,3])

In [61]:
P_value

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

Inference:
    (p-value=0) < (α = 0.05);

Reject Null Hypothesis i.e.
Atleast one sample TAT population mean is different Thus there is variance or difference in average Turn Around 
Time (TAT) of reports of the laboratories on their preferred list.

Q3. Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similaracross regions.

In [65]:
import scipy 

In [66]:
from scipy import stats

In [67]:
from scipy.stats import norm

In [71]:
from scipy.stats import chi2_contingency

In [76]:
Data=pd.read_csv("C:\\Users\\GANESH\\Downloads\\BuyerRatio.csv")

In [77]:
Data

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [78]:
df=Data.iloc[:,1:6]


In [79]:
df

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [81]:
df=np.array([[50,142,131,70],[435,1523,1356,750]])

In [83]:
df

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [85]:
chi2_contingency(df)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

Assume Null hypothesis (Ho) Independence of categorical variables-male, female ratio are similar across region, does not vary and are not related Alternate hyothesis as (Ha) Dependence of categorical variabless male, female are not similar across region, does vary and related

inference p_value=o.6603 > α = 0.05 accept the' Null Hypothesis'

Type Markdown and LaTeX: 𝛼2

 Q4. TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % 
of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. 
The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5%
significance level and help the manager draw appropriate inferences

In [86]:
import scipy

In [87]:
from scipy import stats

In [2]:
from scipy.stats import norm

In [6]:
from scipy.stats import chi2_contingency

In [7]:
import pandas as pd

In [8]:
import numpy as np

In [10]:
DATA=pd.read_csv("C:\\Users\\GANESH\\Downloads\\Costomer+OrderForm.csv")

In [11]:
DATA

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [12]:
DATA.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


In [15]:
DATA.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [16]:
DATA.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [18]:
DATA.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [19]:
DATA.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [21]:
obs=np.array([[271,267,269,280],[29,33,31,20]])

In [22]:
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [25]:
chi2_contingency(obs)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

nference: As (p_value = 0.2771) > (α = 0.05) Accept Null Hypothesis i.e. Independence of categorical variables. customer order forms defective % does not varies by centre

